src/NuGet.Clients/PackageManagement.VisualStudio/SolutionRestore/SolutionRestoreWorker.cs

```diff
@@ -53,7 +59,28 @@ internal sealed class SolutionRestoreWorker : ISolutionRestoreWorker, IDisposabl
             _serviceProvider = serviceProvider;
 
             _componentModel = _serviceProvider.GetService<SComponentModel, IComponentModel>();
+#if VS15
+            _solutionManager = _componentModel.GetService<IVsSolutionManager>();
```

Comment:
First, it is. `VsSolutionManager` class declares MEF export attribute `[Export(typeof(IVsSolutionManager))]`. In your change the instance is resolved explicitly via MEF composition container interface `IComponentModel`.

Then, everywhere else across the NuGet code base the convention was to put all MEF imports in a constructor parameters. See for ex., `VsPackageInstaller`, `VsPackageInstallerServices`, `VsTemplateWizard`, and few more.